client/packages/cli_new/src/commands/bid.ts

```diff
@@ -0,0 +1,50 @@
+import { createCircuitContext } from "@/utils/circuit.ts"
+import { getConfig } from "@/utils/config.ts"
+import { colorLogMsg } from "@/utils/log.ts"
+import { createType } from "@t3rn/types"
+import ora from "ora"
+
+const spinner = ora()
+
+export const handleBidCmd = async (sfxId: string, amount: number) => {
+  const config = getConfig()
+
+  if (!config) {
+    process.exit(1)
+  }
+
+  const { circuit, sdk } = await createCircuitContext()
+  const bidArgs = bid(circuit, sdk, sfxId, amount)
+
+  spinner.start(colorLogMsg("INFO", `Bidding on side effect ${sfxId}...`))
+
+  try {
+    const transaction = circuit.tx.circuit.bidSfx(
+      bidArgs.sfxId,
+      bidArgs.bidAmount
+    )
+    await sdk.circuit.tx.signAndSendSafe(transaction)
+    spinner.stopAndPersist({
+      symbol: "🚩",
+      text: colorLogMsg(
+        "SUCCESS",
+        `Bid successfully placed on side effect ${sfxId} for ${amount}`
+      ),
+    })
+  } catch (error) {
+    spinner.fail(colorLogMsg("ERROR", error))
+    process.exit(1)
+  }
+}
+
+const bid = (
+  circuit: Awaited<ReturnType<typeof createCircuitContext>>["circuit"],
+  sdk: Awaited<ReturnType<typeof createCircuitContext>>["sdk"],
+  sfxId: string,
+  amount: number
+) => {
+  return {
+    sfxId: circuit.createType("SideEffectId", sfxId),
+    bidAmount: createType("u128", sdk.circuit.floatToBn(amount)),
```

Comment:
We have updated the reward system to support other assets to be used for paying the reward, but this was never reflected in the CLI so far. 

The `SideEffect` object contains an optional field `reward_asset_id`, if this is not set the payment is handled in TRN token, and the above code works. If this is not the case, we need to lookup the correct decimals from SDK, to do the `floatToBn`conversion. Otherwise we could end up bidding false amounts.

We have also revamped the token lists in XDNS, so maybe it would make sense to implement this properly in SDK, enabling lookups via asset_id. Lets sync on the one offline, probably a new ticket is needed

Reply:
Could you please create a ticket for this? Let's resolve our conversation so that we can merge this PR seamlessly.

Reply:
As discussed yesterday, this is blocked currently. We need @AwesomeIbex to tell us how we can use xbi asset_registry for this

Reply:
> As discussed yesterday, this is blocked currently. We need @AwesomeIbex to tell us how we can use xbi asset_registry for this

do we already have issue about this?